draft-ietf-moq-transport.md

```diff
@@ -903,37 +903,94 @@ with PUBLISH_ERROR or sets Forward State=0 in PUBLISH_OK. It can also result in
-data stream is already open, it MAY send STOP_SENDING for the data stream along
-with FETCH_CANCEL, but MUST send FETCH_CANCEL.
-
-The Publisher can destroy subscription or fetch state as soon as it has received
-UNSUBSCRIBE or FETCH_CANCEL, respectively. It MUST reset any open streams
-associated with the SUBSCRIBE or FETCH. It can also destroy state after closing
-the FETCH data stream.
+The Publisher can destroy subscription state as soon as it has received
+UNSUBSCRIBE. It MUST reset any open streams associated with the SUBSCRIBE.
 
 The publisher can immediately delete subscription state after sending
-PUBLISH_DONE, but MUST NOT send it until it has closed all related streams. It
-can destroy all FETCH state after closing the data stream.
+PUBLISH_DONE, but MUST NOT send it until it has closed all related streams.
 
 A SUBSCRIBE_ERROR indicates no objects will be delivered, and both endpoints can
 immediately destroy relevant state. Objects MUST NOT be sent for requests that
 end with an error.
 
+### Subscription Filters
+
+The subscriber specifies a filter on the subscription to allow
+the publisher to identify which objects need to be delivered.
+
+All filters have a Start Location and an optional End Group.  Only objects
+published or received via a subscription having Locations greater than or
+equal to Start and strictly less than or equal to the End Group (when
+present) pass the filter.
+
+The `Largest Object` is defined to be the object with the largest Location
+({{location-structure}}) in the track from the perspective of the endpoint
+processing the SUBSCRIBE message. Largest Object updates when the first byte of
+an Object with a larger Location than the previous value is published or
+received through a subscription.
+
+There are 4 types of filters:
+
+Largest Object (0x2): The filter Start Location is `{Largest Object.Group,
+Largest Object.Object + 1}` and `Largest Object` is communicated in
+SUBSCRIBE_OK. If no content has been delivered yet, the filter Start Location is
+{0, 0}. There is no End Group - the subscription is open ended.  Note that due
+to network reordering or prioritization, relays can receive Objects with
+Locations smaller than  `Largest Object` after the SUBSCRIBE is processed, but
+these Objects do not pass the Largest Object filter.
+
+Next Group Start (0x1): The filter Start Location is `{Largest Object.Group + 1,
+0}` and `Largest Object` is communicated in SUBSCRIBE_OK. If no content has been
+delivered yet, the filter Start Location is {0, 0}.  There is no End Group -
+the subscription is open ended. For scenarios where the subscriber intends to
+start from more than one group in the future, it can use an AbsoluteStart filter
+instead.
+
+AbsoluteStart (0x3):  The filter Start Location is specified explicitly in the
+SUBSCRIBE message. The `Start` specified in the SUBSCRIBE message MAY be less
+than the `Largest Object` observed at the publisher. There is no End Group - the
```

Comment:
```suggestion
AbsoluteStart (0x3):  The filter Start Location is specified explicitly in the
SUBSCRIBE message. The `Start` specified in the SUBSCRIBE message MAY be less
than the `Largest Object` observed at the publisher. In the latter case, only the 
objects that are greater than the `Largest Object` is delivered. There is no End Group - the
```

Reply:
Actually, it is not redundant, but it not the agreed to definition of this filter - all objects > Start pass.  Consider the case where the track is a logs and the groups are timestamps.  If I want all log data after time T, I use AbsoluteStart, Start=T.  If the relay has seen T+1, T new objects arriving with T (delayed in network) will still get delivered. If you want to filter based on Largest, use Largest Object.